src/disasm.cpp

```diff
@@ -484,28 +496,44 @@ void jl_strip_llvm_addrspaces(Module *m)
     PM.run(*m);
 }
 
+
+#include <iostream>
+#include "llvm/IR/InstIterator.h"
+#include <sstream>
 // print an llvm IR acquired from jl_get_llvmf
 // warning: this takes ownership of, and destroys, f->getParent()
-extern "C" JL_DLLEXPORT
+extern "C" JL_DLLEXPORT 
 jl_value_t *jl_dump_function_ir_impl(void *f, char strip_ir_metadata, char dump_module, const char *debuginfo)
 {
     std::string code;
     raw_string_ostream stream(code);
+    std::vector<std::pair<int, std::string>> ccalls;
 
+    bool annotate_ccalls = strip_ir_metadata; // do not annotate when raw = true
     {
         Function *llvmf = dyn_cast_or_null<Function>((Function*)f);
+
         if (!llvmf || (!llvmf->isDeclaration() && !llvmf->getParent()))
             jl_error("jl_dump_function_ir: Expected Function* in a temporary Module");
-
+        int inst_count = 0;
+        if (annotate_ccalls) { // find all ccall instructions to annotate
+        for (inst_iterator I = inst_begin(llvmf), E = inst_end(llvmf); I != E; ++I) {
+            MDNode* md = I->getMetadata("julia.ccall.name");
+                if (md != nullptr) {
+                    std::pair<int, std::string> p(inst_count++, std::string(cast<MDString>(md->getOperand(0))->getString().data()));
```

Comment:
This kind of looks like `ccalls[i].first == i` for all elements in ccalls, because we increment inst_count only when we append to ccalls. Can we just turn this from a vector<pair<int, string>> to a vector<string> and have the index implicitly tracked?

Reply:
Nice catch! I should be incrementing inst_count every time as that is how I access in emitInstructionAnnot, fixed

Reply:
Also, if we're not deleting it can we have ccalls be a map upfront instead of making a vector of pairs and then later turning it into a map?